iModelCore/ECPresentation/Source/Shared/ECExpressions/ECExpressionContextsProvider.cpp

```diff
@@ -49,11 +49,35 @@ struct CommonRulesEngineSymbolsProvider : IECSymbolProvider
         evalResult.SetValueList(*IValueListResult::Create(values));
         return ExpressionStatus::Success;
         }
+
+    /*---------------------------------------------------------------------------------**//**
+    * @bsimethod
+    +---------------+---------------+---------------+---------------+---------------+------*/
+    static ExpressionStatus GuidToStr(EvaluationResult& evalResult, void*, EvaluationResultVector& args)
+        {
+        bvector<EvaluationResult> values;
+        if (args.size() == 0 || args.size() > 1)
+            return ExpressionStatus::WrongNumberOfArguments;
+
+        EvaluationResultCR arg = args[0];
+        if (!arg.IsECValue() && !arg.GetECValue()->IsBinary())
+            ExpressionStatus::IncompatibleTypes;
+
+        size_t guidSize = sizeof(BeGuid);
+        BeGuid* guid = (BeGuid*)arg.GetECValue()->GetBinary(guidSize);
+
+        if (!guid->IsValid())
+            return ExpressionStatus::IncompatibleTypes;

```

Comment:
No sure this is right. Need a test that tries to convert an empty guid to string to confirm.